web/client/components/data/featuregrid/filterRenderers/baseDateTimeFilter/BaseDateTimeFilter.js

```diff
@@ -1,31 +1,34 @@
- /**
-  * Copyright 2017, GeoSolutions Sas.
-  * All rights reserved.
-  *
-  * This source code is licensed under the BSD-style license found in the
-  * LICENSE file in the root directory of this source tree.
-  */
+/**
+ * Copyright 2017, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
 
-const React = require('react');
-const PropTypes = require('prop-types');
-require('react-widgets/lib/less/react-widgets.less');
-const {DateTimePicker} = require('react-widgets');
+import React from 'react';
+import PropTypes from 'prop-types';
+import {intlShape} from 'react-intl';
+import DateTimePicker from '../../../../misc/datetimepicker';
+import LocaleUtils from '../../../../../utils/LocaleUtils';
+import { getDateTimeFormat } from '../../../../../utils/TimeUtils';
+import AttributeFilter from '../AttributeFilter';
+import utcDateWrapper from '../../../../misc/enhancers/utcDateWrapper';
+import 'react-widgets/lib/less/react-widgets.less';
 
-const LocaleUtils = require('../../../../utils/LocaleUtils');
-const {getDateTimeFormat} = require('../../../../utils/TimeUtils');
-const AttributeFilter = require('./AttributeFilter');
-const utcDateWrapper = require('../../../misc/enhancers/utcDateWrapper');
 const UTCDateTimePicker = utcDateWrapper({
     dateProp: "value",
     dateTypeProp: "type",
     setDateProp: "onChange"
 })(DateTimePicker);
 
+
 class DateFilter extends AttributeFilter {
     static propTypes = {
         type: PropTypes.string,
         disabled: PropTypes.boolean,
-        onChange: PropTypes.func
+        onChange: PropTypes.func,
+        intl: intlShape.isRequired
```

Comment:
Please find out a way to wrap i18n in a separate  HOC outside, to allow reuse of DateTiimeFilter (and DateTimePicker) and avoid to redefine everything.